test/integration/users.test.js

```diff
@@ -567,4 +572,63 @@ describe("Users", function () {
         });
     });
   });
+
+  describe("PATCH /users/add-default-archived-role", function () {
+    let nonSuperUserId;
+    beforeEach(async function () {
+      nonSuperUserId = await addUser(nonSuperUser);
+      await addUser(userRolesDoesNotExists);
+      await addUser(userArchivedRoleDoesNotExists);
+      await addUser(userAlreadyArchived);
+    });
+
+    it("Should return 401 if user is not a super user", function (done) {
+      const nonSuperUserJwt = authService.generateAuthToken({ userId: nonSuperUserId });
+      chai
+        .request(app)
+        .patch(`/users/add-default-archived-role`)
+        .set("cookie", `${cookieName}=${nonSuperUserJwt}`)
+        .end((err, res) => {
+          if (err) {
+            return done(err);
+          }
+
+          expect(res).to.have.status(401);
+          expect(res.body).to.be.a("object");
+          expect(res.body.message).to.equal("You are not authorized for this action.");
+
+          return done();
+        });
+    });
+
+    it("Should add default archived role to user only where it does not exists", function (done) {
+      chai
+        .request(app)
+        .patch(`/users/add-default-archived-role`)
```

Comment:
Can we have a better API name here please.

API names should be nouns, not verbs.

Reply:
Yes. This was a miss on my part. Will think proper naming from next time onwards.